src/blocks/mrc_get_parameter.ts

```diff
@@ -70,11 +69,73 @@ const GET_PARAMETER_BLOCK = {
     this.mrcHasWarning = false;
 
     this.setStyle(MRC_STYLE_VARIABLES);
+    
+    // Create a custom dropdown that accepts any value and displays it correctly
+    class CustomParameterDropdown extends Blockly.FieldDropdown {
+      override doClassValidation_(newValue?: string): string | null {
+        // Always accept the value, even if it's not in the current options
+        return newValue ?? null;
+      }
+      
+      override getText_(): string {
+        // Always return the current value, even if not in options
+        return this.value_ || '';
+      }
+    }
+    
+    const blockRef = this;
+    // Use a dummy initial option - it will be replaced when setValue is called
+    const dropdown: Blockly.Field = new CustomParameterDropdown(
+      function() {
+        // This function will be called to regenerate options when dropdown opens
+        return blockRef.getParameterOptions();
+      }
+    );
+    
+    dropdown.setValidator(this.validateParameterSelection.bind(this));
+    
     this.appendDummyInput()
         .appendField(Blockly.Msg.PARAMETER)
-        .appendField(createFieldNonEditableText(''), FIELD_PARAMETER_NAME);
+        .appendField(dropdown, FIELD_PARAMETER_NAME);
     this.setOutput(true, this.mrcParameterType);
   },
+  getParameterOptions: function(this: GetParameterBlock): [string, string][] {
+    const legalParameterNames: string[] = [];
```

Comment:
Is "legal" the right word here? Does this array contain "legal" names, or the method/event's parameter names?
If you can think of a better name, great. Otherwise, please add a comment explaining what "legal" means in this context.